src/Microsoft.DotNet.ProjectJsonMigration/transforms/IncludeContextTransform.cs

```diff
@@ -68,18 +83,21 @@ internal class IncludeContextTransform : ConditionalTransform<IncludeContext, IE
 
         private readonly string _itemName;
         private bool _transformMappings;
+        private string[] _patternsBlackList;
         private bool _emitBuiltInIncludes;
         private readonly List<ItemMetadataValue<IncludeContext>> _metadata = new List<ItemMetadataValue<IncludeContext>>();
 
         public IncludeContextTransform(
             string itemName,
             bool transformMappings = true,
             Func<IncludeContext, bool> condition = null,
-            bool emitBuiltInIncludes = true) : base(condition)
+            bool emitBuiltInIncludes = true,
+            string[] patternsBlackList = null) : base(condition)
         {
             _itemName = itemName;
             _transformMappings = transformMappings;
             _emitBuiltInIncludes = emitBuiltInIncludes;
+            _patternsBlackList = patternsBlackList;
```

Comment:
In order to avoid the null checks, make the field readonly and do a null coalesce on assignment:

```
_patternsBlackList = patternsBlackList ?? new string[0];
```